Makefile

```diff
@@ -169,11 +166,29 @@ ffac-patch:
 	fi
 	@$(GLUON_GIT) branch -M patched
 
+.cmp-git-head: FORCE | ffac-patch
+	@$(GLUON_GIT) rev-parse @{0} | cmp -s '$@' || $(GLUON_GIT) rev-parse @{0} > '$@'
+
+.modules: release.mk modules .cmp-git-head $(PATCH_DIR) $(PATCH_FILES) | ffac-patch
+	@echo
+	@echo Updating Gluon modules…
+	@rm -f .modules
+	+$(GLUON_MAKE) update
+	@if [ -f "$(OPKG_KEY_FOLDER)/key-build" ] && [ ! -f "$(GLUON_BUILD_DIR)/openwrt/key-build" ]; then \
+		echo 'Installing our opkg keys'; \
+		cp $(OPKG_KEY_FOLDER)/key-build* $(GLUON_BUILD_DIR)/openwrt/; \
+	fi
+	@touch .modules
+
 gluon-clean:
+	rm -f .modules
 	rm -rf $(GLUON_BUILD_DIR)
+	@echo
 
 output-clean:
 	mkdir -p output/
 	rm -rf output/*
 
 clean: gluon-clean output-clean
+
+FORCE: ;
```

Comment:
what ist this?

Reply:
as explained in the commit message (iirc):
FORCE is the old .PHONY
with the big difference, that FORCE rebuilds everytime

.PHONY appears to have an exception for files when all requirements didn't need an update (rebuild), no clue
.cmp-git-hash only works with a FORCE target

FORCE is just a name everyone uses by convention

.cmp-git-hash is just an implementation of .up-to-date, I got it from here
https://stackoverflow.com/a/24328410
it's all explained here. also why we need FORCE instead of .PHONY